### PR TITLE
feat: add smoketest for cacerts(check alias number)

### PR DESCRIPTION
**Pull request by zdtsw**
- add new smoketest for cacerts into CommonFeatureTests
- only enable matching cacerts number at the moment
Positive case: https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk19/job/jdk19-mac-x64-temurin_SmokeTests/17/console
Negative case (when we had problem with cacerts unfixed):  https://ci.adoptopenjdk.net/view/Failing%20Temurin%20jobs/job/build-scripts/job/jobs/job/jdk/job/jdk-linux-arm-temurin_SmokeTests/37/console

Ref test case from [jdk19](https://github.com/adoptium/jdk19/blob/master/test/jdk/sun/security/lib/cacerts/VerifyCACerts.java)
Ref: https://github.com/adoptium/temurin-build/issues/3034

### CODE REVIEW

File: test/functional/buildAndPackage/src/net/adoptium/test/JdkVersion.java [hunk 1]
```diff
@@ -145,6 +145,14 @@ public int getPatch() {
         return patch;
     }
 
+    /**
+     * Retrieve value of JAVA_HOME.
+     * @return String   JAVA_HOME
```
Comment (karianna) [inline test/functional/buildAndPackage/src/net/adoptium/test/JdkVersion.java:150]:
```suggestion
     * @return String - value of environment variable JAVA_HOME
```

File: test/functional/buildAndPackage/src/net/adoptium/test/VerifyCACertsTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        put("digicerttrustedrootg4 [jdk]",  "55:2F:7B:DC:F1:A7:AF:9E:6C:E6:72:01:7F:4F:12:AB:F7:72:40:C7:8E:76:1A:C2:03:D1:D9:D2:0A:C8:99:88");
+        put("digicertassuredidrootca [jdk]","3E:90:99:B5:01:5E:8F:48:6C:00:BC:EA:9D:11:1E:E7:21:FA:BA:35:5A:89:BC:F1:DF:69:56:1E:3D:C6:32:5C");
+        put("digicertassuredidg2 [jdk]",    "7D:05:EB:B6:82:33:9F:8C:94:51:EE:09:4E:EB:FE:FA:79:53:A1:14:ED:B2:F4:49:49:45:2F:AB:7D:2F:C1:85");
+        put("digicertassuredidg3 [jdk]",    "7E:37:CB:8B:4C:47:09:0C:AB:36:55:1B:A6:F4:5D:B8:40:68:0F:BA:16:6A:95:2D:B1:00:71:7F:43:05:3F:C2");
+        put("digicerthighassuranceevrootca [jdk]", "74:31:E5:F4:C3:C1:CE:46:90:77:4F:0B:61:E0:54:40:88:3B:A9:A0:1E:D0:0B:A6:AB:D7:80:6E:D3:B1:18:CF");
+    }
+
+    private static final HexFormat HEX = HexFormat.ofDelimiter(":").withUpperCase();
+    */
+    private static final String JHOME = getJavaHome();
```
Comment (karianna) [inline test/functional/buildAndPackage/src/net/adoptium/test/VerifyCACertsTest.java:60]:
```suggestion
    private static final String JAVA_HOME = getJavaHome();
```

File: test/functional/buildAndPackage/src/net/adoptium/test/VerifyCACertsTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    // concat absolute path regardless OS
+    private static final String CACERTS = JHOME + File.separator + "lib" + File.separator + "security" + File.separator + "cacerts";
```
Comment (karianna) [inline test/functional/buildAndPackage/src/net/adoptium/test/VerifyCACertsTest.java:62]:
```suggestion
    private static final String CACERTS = JAVA_HOME + File.separator + "lib" + File.separator + "security" + File.separator + "cacerts";
```

File: test/functional/buildAndPackage/src/net/adoptium/test/VerifyCACertsTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+     */
+    @Test
+    public void isCertNrMatch() throws Exception {
+        LOGGER.info("cacerts file: " + CACERTS);
+        try {
+            byte[] data = Files.readAllBytes(Path.of(CACERTS));
+
+            KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType()); // fallback to JSK if not set defaulttype
+            ks.load(new ByteArrayInputStream(data), "changeit".toCharArray());
+            assertEquals(ks.size(), EXPCOUNT, "Failed to match CA certs number.");
```
Comment (karianna) [inline test/functional/buildAndPackage/src/net/adoptium/test/VerifyCACertsTest.java:75]:
```suggestion
            assertEquals(ks.size(), EXPECTED_COUNT, "Failed to match CA certs number.");
```

**Review by karianna — CHANGES_REQUESTED** [review]
early feedback

**Comment by zdtsw** [comment]
one failed on alpine jdk19 because of env. timeout

**Review by andrew-m-leonard — APPROVED** [review]
Looks good i think

**Comment by andrew-m-leonard** [comment]
@zdtsw probably worth re-running the failed CI tests

**Comment by zdtsw** [comment]
> @zdtsw probably worth re-running the failed CI tests

yeah, it is on going, will take a while for the result

**Pull request merged**